scripts/build-json/build-link-lists.js

```diff
@@ -85,23 +164,23 @@ function linkListFromChapterList(
  *
  * Each of those directories is expected to contain content for a single page.
  */
-function linkListFromDirectory(
-  title,
-  directory,
-  includeShortDescriptions = false
-) {
+function linkListFromDirectory(title, directory, foreach) {
   const fullPath = path.join(ROOT, directory);
   let itemDirectories = fs
     .readdirSync(fullPath, { withFileTypes: true })
     .filter(item => item.isDirectory());
   itemDirectories = itemDirectories.map(itemDirectory =>
     path.join(fullPath, itemDirectory.name)
   );
+
+  const listContent = [];
+  for (const dirEntry of itemDirectories) {
+    listContent.push(buildLinkItem(dirEntry, foreach));
+  }
+
   return {
     title: title,
-    content: itemDirectories.map(
-      itemFromDirectory.bind(null, includeShortDescriptions)
-    )
+    content: listContent
```

Comment:
Well... I wanted to pass `foreach` into `buildLinkItem()` when I was calling it from `linkListFromDirectory()`.

So because I was calling `buildLinkItem()` indirectly via `map`, I was [using `bind()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/bind#Partially_applied_functions) to smuggle the `foreach` argument into `buildLinkItem()`.

But that seems quite obscure really. And the bug I found  - that `resolveListSpec()` was modifying the existing unresolved list spec, instead of creating a new one, which meant that things got weird when you try to resolve the list spec for the next entry - was harder to find that it would have been, because of this. I think using `for...of` here makes it more obvious what's going on.

But maybe there's a better way to do this?

Reply:
Thanks Daniel, for the suggestion. I've made that change!